### ISSUE TITLE
Reloading a DSL-rule-file can mess up other rule files/the rule engine

### ISSUE THREAD

**Issue by Flole998** [openhab/openhab-core#2942]
I changed a rule file and an item file while OH was running and OH correctly picked up the changes:

```
2022-04-30 15:33:28.526 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'default.items'
2022-04-30 15:34:28.598 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'other.rules'
```
but from this point onwards something in the rule engine broke, causing one of my rules to fail:

```
2022-04-30 15:43:14.216 [ERROR] [internal.handler.ScriptActionHandler] - Script execution of rule with UID 'myrules-3' failed: Could not cast OFF to org.openhab.core.library.types.DecimalType; line 54, column 21, length 79 in myrules
```

the line that is failing is

```
val lastdaymax = (Temperature_Outdoor.maximumSince(now.minusDays(1),"rrd4j").state as DecimalType)
```

and normally this works perfectly fine, however after the changes for some reason the state returned OFF (that Item is a Number and not a Switch). I have seen other cases where the rule engine ended up in a somewhat broken state, but I never really figured out why that happened. After a restart everything is back to normal again. Any idea on how to further look into this (or even where the issue could be) would be appreciated.

**Comment by J-N-K** [comment]
That seems to be a bug in the persistence extensions. I‘ll have a look. Can you reproduce the issue? If so, can you send me the rrd database of that item to github@klug.nrw?

**Comment by Flole998** [comment]
I can't reproduce it, now that I restarted openHAB the issue is gone again. Even triggering a reload of the items and rules file does not cause it again. I will try to see if I can reproduce it if I try to reload those files a few times, but I highly doubt it.

**Comment by J-N-K** [comment]
Is this a Number or a Number:Temperature item?

**Comment by Flole998** [comment]
It's a number:
```
Number                            Temperature_Outdoor                           "Außentemperatur [%.2f °C]"                                           <temperature>        (Weather, Weather_Chart, RRDPersistence)                                                                                     {channel="mqtt:topic:localBroker:XX:Temperature", channel="mqtt:topic:localBroker:XX:OutTemp" [profile="follow"], alexa="TemperatureSensor.temperature" [scale="Celsius"]}
```

**Comment by J-N-K** [comment]
Did you succeed in reproducing this issue?

**Comment by Flole998** [comment]
Nope, I tried a few times but this issue never showed up again. Any idea what I could look into in case I ever see this (or something similar) again?

**Comment by J-N-K** [comment]
Immediately take a copy of the rrd database. I have the slight feeling that the issue is related to the values returned by the rrd persistence service. Since a the moment it seems to be a single event, I would suggest to close this issue here and re-open if it occurs again.

**Comment by Flole998** [comment]
I don't think it's RRD related as a restart fixed it and I saw the message multiple times before and nothing but a restart fixed it, I would have expected that if it's a RRD issue then it wouldn't have been fixed by a restart. Maybe I can try to see if reloading the persistence service fixes it or restarting of one of the bundles with "Rule" in it's name?

Anyways, closing this until I see this or something similar again and can provide additional information.
